### ISSUE TITLE
ImportError: No module named '_signal'

### ISSUE THREAD

**Issue by niemasd** [nx-python/PyNX#9]
Can trigger with the following:

```python
import signal
```

Error:

```bash
  File "/switch/PyNX/lib/python3.5/signal.py", line 1, in <module>
    import _signal
ImportError: No module named '_signal'
```

**Comment by elnardu** [comment]
As I recall switch's os does not have signals, so there is no point in
using signal module

**Comment by niemasd** [comment]
I was trying to set up `pyftpdlib` to try to have a PyNX-backed FTP server

**Comment by elnardu** [comment]
You will have to manually remove every usage of subprocess and signal modules in the source

**Comment by rituzangle** [comment]
It might still help someone:
# Debugging Python: `ModuleNotFoundError: No module named '_signal'`
TL;DR - Know your Python. 
Error popped up on RustPython. Until I realized that all sorts of 'fixes' were useless.
❯ python --version
Python 3.11.0 (RustPython)

This document explains how to diagnose and fix the `ModuleNotFoundError: No module named '_signal'` error in Python. This error is not caused by incorrect code syntax, but by the Python environment itself.

## The Problem

You attempt to run a Python script that imports a standard library module, such as `subprocess`, and you encounter the following traceback:

```
Traceback (most recent call last):
  File "your_script.py", line 1, in <module>
    import subprocess
  File "/path/to/python/subprocess.py", line 49, in <module>
    import signal
  File "/path/to/python/signal.py", line 1, in <module>
    ModuleNotFoundError: No module named '_signal'
```

Despite `import subprocess` being the correct syntax, the program crashes.

## The Root Cause: Environment vs. Syntax

The key to understanding this error is recognizing that "Python" is a language specification, and there are multiple *implementations* (interpreters) of it.

1.  **CPython (The Standard):** This is the official and most widely-used Python interpreter, written in the C programming language. Its standard library often relies on C-level modules for performance and to interact with the operating system. The `subprocess` module, for example, depends on a low-level module named `_signal` to manage system processes.

2.  **Alternative Interpreters (e.g., RustPython, Jython, IronPython):** These are other implementations of the Python language written in different languages (Rust, Java, C#, respectively). While they are designed to be compatible with the Python language, they may not have a complete implementation of the CPython standard library, especially the C-level components.

The `ModuleNotFoundError: No module named '_signal'` error occurs because the Python interpreter you are using (in this case, **RustPython**) was not built with the `_signal` module, which is a necessary dependency for the `subprocess` library to function.

**In short, your code was correct, but the environment was incomplete.**

## The Solution

The solution is to ensure you are running your script with a standard CPython interpreter that includes the full standard library.

### 1. Verify Your Python Interpreter

You can check the version and type of your Python interpreter by running:

```bash
python --version
# Or for Python 3
python3 --version
```

If the output indicates a non-standard interpreter (e.g., `Python 3.11.0 (RustPython)`), you have found the source of the problem.

### 2. Switch to a Standard CPython Environment

The most reliable fix is to run your script with a standard CPython installation.

- **Using System Python:** Most operating systems (like macOS and Linux distributions) come with a pre-installed version of CPython. You can often invoke it directly:
  ```bash
  /usr/bin/python3 your_script.py
  ```

- **Using a Version Manager (Recommended):** Tools like `pyenv`, `asdf`, or `conda` allow you to install and manage multiple versions of Python. This is the best practice for avoiding conflicts.
  ```bash
  # Example using pyenv to set a standard Python version
  pyenv install 3.11.5
  pyenv local 3.11.5

  # Now, running python will use the CPython version
  python your_script.py
  ```

### 3. (Context-Specific) Find an Alternative Script

In some cases, a project may contain equivalent scripts in different languages. In the situation that prompted this document, a `cleanup-environment.js` file existed alongside the failing `cleanup-environment.py`. Running the JavaScript version with Node.js was a practical way to achieve the goal without debugging the Python environment.

## Key Takeaway

If you encounter a `ModuleNotFoundError` for a module that seems like it should be standard (especially one with a leading underscore like `_signal`), the first thing to suspect is the Python environment itself. Verify you are using a standard CPython interpreter, as alternative implementations may have an incomplete standard library.

**Comment by DavidBuchanan314** [comment]
I don't know why you would post LLM slop in a 7 year old closed issue about an entirely different python runtime.
